### PR TITLE
Integrate Dokka to generate markdown docs

### PR DESCRIPTION
**Pull request by ShaishavGandhi**
This generates markdown documentation for better compatibility with mkdocs.

### CODE REVIEW

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -22,6 +22,7 @@ def versions = [
     jmhLib: '1.21',
     jmhPlugin: '0.4.8',
     nullawayPlugin: '0.1',
+    dokka: '0.9.17',
```
Comment (ZacSweers) [inline gradle/dependencies.gradle]: Why not 0.9.18?

File: rxdogtag-autodispose/build.gradle [hunk 2]
```diff
@@ -29,4 +30,15 @@ dependencies {
...
 }
 
+dokka {
+  externalDocumentationLink {
+    url = new URL("http://reactivex.io/RxJava/2.x/javadoc/")
+  }
+
+  outputFormat = 'gfm'
+  outputDirectory = "$rootDir/docs/0.x"
+  classpath = new ArrayList<File>(project.tasks['assemble'].outputs.files.files)
```
Comment (ZacSweers) [inline rxdogtag-autodispose/build.gradle]: This can just be the groovy `[]` initializer

File: rxdogtag-autodispose/build.gradle [hunk 3]
```diff
@@ -29,4 +30,15 @@ dependencies {
   testImplementation project(':rxdogtag').sourceSets.test.output
 }
 
+dokka {
+  externalDocumentationLink {
+    url = new URL("http://reactivex.io/RxJava/2.x/javadoc/")
+  }
+
+  outputFormat = 'gfm'
```
Comment (ZacSweers) [inline rxdogtag-autodispose/build.gradle]: Can you leave a comment why this format was needed?

File: gradle/dependencies.gradle (hunk 1 above)
Comment (ShaishavGandhi) [reply gradle/dependencies.gradle]: Didn't know. 7a53222

File: rxdogtag-autodispose/build.gradle (hunk 3 above)
Comment (ShaishavGandhi) [reply rxdogtag-autodispose/build.gradle]: 7a53222

File: rxdogtag-autodispose/build.gradle (hunk 2 above)
Comment (ShaishavGandhi) [reply rxdogtag-autodispose/build.gradle]: It wasn't able to cast it to a List<File>.

**Pull request merged**